resources/osl_heisenbridge.rb

```diff
@@ -0,0 +1,61 @@
+# osl_heisenbridge
+#
+# Heisenbridge is an IRC bouncer for a Matrix server
+
+resource_name :osl_heisenbridge
+provides :osl_heisenbridge
+unified_mode true
+
+default_action :create
+
+property :container_name, String, name_property: true
+property :port, Integer, default: 9898
+property :matrix_host_resource, Symbol, required: true, default: :osl_synapse
+property :matrix_host_domain, String, required: true
+
+action :create do
+  # Get the required information taken from the matrix host resource
+  matrix_resource = find_resource(new_resource.matrix_host_resource, new_resource.matrix_host_domain)
+  path = matrix_resource.path
+  name = matrix_resource.name
+  network = matrix_resource.network
```

Comment:
I don't think this should be automated like this. This is bound to create issues later. When we deploy this in production, we will know all of these so we should just have properties in this resource to set it.